### PR TITLE
Queries day 2

### PR DESCRIPTION
**Pull request by NatanimA**
**In this PR:**
- Add a column species of type string to your animal's table. Modify your schema.sql file.
```
ALTER TABLE animals
ADD COLUMN species VARCHAR(40);
```

- Insert the following data:
-  His name is Charmander. He was born on Feb 8th, 2020, and currently weighs 11kg. He is not neutered and he has never tried to escape.
`INSERT into animals(name,date_of_birth,weight_kg,neutered,escape_attempts) values('Charmander','2020-Feb-8',11,false,0);
`
- Her name is Plantmon. She was born on Nov 15th, 2021, and currently weighs 5.7kg. She is neutered and she has tried to escape 2 times.
`INSERT into animals(name,date_of_birth,weight_kg,neutered,escape_attempts) values('Plantmon','2021-Nov-15',5.7,true,2);
`
- His name is Squirtle. He was born on Apr 2nd, 1993, and currently weighs 12.13kg. He was not neutered and he has tried to escape 3 times.
`INSERT into animals(name,date_of_birth,weight_kg,neutered,escape_attempts) values('Squirtle','1993-Apr-2',12.13,false,3);
`
- His name is Angemon. He was born on Jun 12th, 2005, and currently weighs 45kg. He is neutered and he has tried to escape once.
`INSERT into animals(name,date_of_birth,weight_kg,neutered,escape_attempts) values('Angemon','2005-Jun-12',45,true,1);
`
- His name is Boarmon. He was born on Jun 7th, 2005, and currently weighs 20.4kg. He is neutered and he has tried to escape 7 times.
`INSERT into animals(name,date_of_birth,weight_kg,neutered,escape_attempts) values('Boarmon','2005-Jun-7',20.4,true,7);
`
- Her name is Blossom. She was born on Oct 13th, 1998, and currently weighs 17kg. She is neutered and she has tried to escape 3 times.
`INSERT into animals(name,date_of_birth,weight_kg,neutered,escape_attempts) values('Blossom','1998-Oct-13',17,true,3);
`
- His name is Ditto. He was born on May 14th, 2022, and currently weighs 22kg. He is neutered and he has tried to escape 4 times.
`INSERT into animals(name,date_of_birth,weight_kg,neutered,escape_attempts) values('Ditto','2022-May-14',22,true,4);
`
- Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.

- Using Rollback

- Inside a transaction:
- Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
- Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
- Commit the transaction.

- Verify that change was made and persists after commit.

- Now, take a deep breath and... Inside a transaction delete all records in the animals table, then roll back the transaction.
- After the rollback verify if all records in the animals table still exists. After that, you can start breathing as usual ;)

- Inside a transaction:
- Delete all animals born after Jan 1st, 2022.
- Create a savepoint for the transaction.
- Update all animals' weight to be their weight multiplied by -1.
- Rollback to the savepoint
- Update all animals' weights that are negative to be their weight multiplied by -1.
- Commit transaction

- Write queries to answer the following questions:
- How many animals are there?

- How many animals have never tried to escape?

- What is the average weight of animals?

- Who escapes the most, neutered or not neutered animals?

- What is the minimum and maximum weight of each type of animal?

- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
last

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -8,3 +8,28 @@ INSERT into animals(name,date_of_birth,weight_kg,neutered,escape_attempts) value
 -- Her name is Devimon. She was born on May 12th, 2017, and currently weighs 11kg. She is neutered and she has tried to escape 5 times.
 INSERT into animals(name,date_of_birth,weight_kg,neutered,escape_attempts) values('Devimon','2017-May-12',11,true,5);
 
+-- Add a column species of type string to your animals table. Modify your schema.sql file.
+ALTER TABLE animals
+ADD COLUMN species VARCHAR(40);
```
Comment (Munsa1) [inline data.sql]: - [ ] This statement should be in the `schema.db` file and not the `data.db` file, kinly fix this

File: queries.sql [hunk 2]
```diff
@@ -14,4 +14,90 @@ select * from animals where neutered = true;
 -- Find all animals not named Gabumon.
 select * from animals where name != 'Gabumon';
 -- Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)
-select * from animals where weight_kg >= 10.4 and weight_kg <= 17.3 ;
\ No newline at end of file
+select * from animals where weight_kg >= 10.4 and weight_kg <= 17.3 ;
+
+-- Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
+UPDATE animals
+SET
```
Comment (Munsa1) [inline queries.sql]: - [ ] Before starting this transaction, a `begin;` statement is required kindly aad it above line 20

File: queries.sql [hunk 3]
```diff
@@ -14,4 +14,90 @@ select * from animals where neutered = true;
...
+
+-- Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
+UPDATE animals
+SET 
+species = 'unspecified';
+
+UPDATE animals
+SET 
+species = 'unspecified';
+ROLLBACK;
```
Comment (Munsa1) [inline queries.sql:25]: - [ ] Before and after the `rollback` statement, its required that you check the changes were made by selecting the species from the animal table. Kindly add that select statement above and below line 27.
 e.g `SELECT species FROM animals;`

File: queries.sql [hunk 4]
```diff
@@ -14,4 +14,90 @@ select * from animals where neutered = true;
...
+WHERE name like '%mon';
+
+-- Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+
+-- Commit the transaction.
+
+COMMIT;
```
Comment (Munsa1) [inline queries.sql:42]: - [ ] Again here before commiting the changes verify that the changes where made by using the select statement as you have done in line 46.

File: queries.sql [hunk 5]
```diff
@@ -14,4 +14,90 @@ select * from animals where neutered = true;
...
+DELETE from animals;
+ROLLBACK;
+SELECT * from animals;
+
+-- Inside a transaction
+BEGIN;
+SAVEPOINT BEFORE_DELETE;
+-- Delete all animals born after Jan 1st, 2022.
+DELETE from animals
+where date_of_birth > '2022-1-1';
```
Comment (Munsa1) [inline queries.sql]: -  [ ] Here you should delete animals where date of birth is greater than or equal to `2022-1-1`, kindly add the `=` sign so you have `>=` sign which produces a different result

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @NatanimA , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Correct Gitflow
## Required Changes ♻️
_Check the comments under the review._
- [ ] Kindly use the new Readme file [template](https://github.com/microverseinc/readme-template/blob/master/README.md)
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Nemwel-Boniface — APPROVED** [review]
## Hello @NatanimA      :wave: ,
I am Nemwel your reviewer for this PR :man_beard:  

Thank you for meeting all the project requirements good job! :partying_face: 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! :partying_face: 🎉 



## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️ 
Met all the milestone requirements ✔️ 

## [Optional] suggestions

- [ ] The project was supposed to be invalidated by the previous reviewer, however since it was not kindly ensure that you use the [default README template](https://github.com/microverseinc/readme-template) provided to you by microverse to create your README.md and include the headers

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**